sklearn/model_selection/tests/test_search.py

```diff
@@ -1725,3 +1725,49 @@ def test_deprecated_grid_search_iid():
     grid = GridSearchCV(SVC(gamma='scale'), param_grid={'C': [1]}, cv=KFold(2))
     # no warning because no stratification and 54 % 2 == 0
     assert_no_warnings(grid.fit, X, y)
+
+
+def test_empty_cv_iterator_error():
+    # Generate sample data
+    X, y = make_blobs(n_samples=54, random_state=0, centers=2)
```

Comment:
`X` and `y` are actually defined globally in this file. You don't need to create them at all, you can just use them.